crates/sncast/src/lib.rs

```diff
@@ -302,6 +310,23 @@ pub async fn wait_for_tx(
     ))
 }
 
+async fn get_revert_reason(
+    provider: &JsonRpcClient<HttpTransport>,
+    tx_hash: FieldElement,
+) -> Result<&str> {
+    match provider.get_transaction_receipt(tx_hash).await {
+        Ok(receipt) => match receipt.execution_result() {
+            starknet::core::types::ExecutionResult::Succeeded => {
+                unreachable!()
```

Comment:
why `unreachable`?

Reply:
We use this function to get the reason for the revert, so currently, it's executed only when we know that tx was reverted.

Reply:
but what we should do in case when the status doesn't match `Reverted`? should we panic?

Reply:
yes, this is a error in our codebase, so we should panic

Reply:
> And what about the case when a call response with an error?
throw this error, but do not panic

Reply:
done